presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -69,6 +69,7 @@
         implements ConnectorSplitManager
 {
     public static final String PRESTO_OFFLINE = "presto_offline";
+    private static final String STRUCT_PREFIX = "struct<";
```

Comment:
Yeah, this part pained me pretty bad. I dug around for something more type safe but I missed HiveType. Refactoring.